Wvlet CLI (wv): Claude Code support

Currently, wv works as a command line tool, but there seems to be some issues when running it from Claude Code:

Commands:
```
wv -c (query)
wv --file (query) 
```

From Claude Code:
```
  ⎿ 2025-10-20 12:46:15.611+0900 error [WvletREPLMain]
    java.lang.IllegalStateException: Unable to create a terminal
    	at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:707)
    	at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:672)
    	at wvlet.lang.cli.WvletREPL.<init>(WvletREPL.scala:54)
    	at wvlet.lang.cli.WvletREPLMain.$anonfun$51(WvletREPLMain.scala:85)
    	at wvlet.airframe.surface.ObjectFactory$$anon$1.newInstance(Surfaces.scala:73)
    	at wvlet.airframe.AirframeSession.buildInstance(AirframeSession.scala:449)
    	at wvlet.airframe.AirframeSession.buildInstance$$anonfun$3(AirframeSession.scala:414)
    	at scala.Option.getOrElse(Option.scala:201)
    	at wvlet.airframe.AirframeSession.buildInstance(AirframeSession.scala:415)
    	at wvlet.airframe.AirframeSession.$anonfun$15(AirframeSession.scala:335)
    	at wvlet.airframe.AirframeSession.getOrBuildSingleton(AirframeSession.scala:101)
    	at wvlet.airframe.AirframeSession.getInstance(AirframeSession.scala:336)
    	at wvlet.airframe.AirframeSession.get(AirframeSession.scala:178)
    	at wvlet.lang.cli.WvletREPLMain.repl$$anonfun$2(WvletREPLMain.scala:97)
    	at wvlet.airframe.Design.runWithSession(Design.scala:201)
    	at wvlet.airframe.Design.withSession(Design.scala:214)
    	at wvlet.lang.cli.WvletREPLMain.repl(WvletREPLMain.scala:97)
    	at wvlet.lang.cli.WvletREPLMain$.$anonfun$152(WvletREPLMain.scala:20)
    	at wvlet.airframe.surface.ClassMethodSurface.call$$anonfun$1(Surfaces.scala:430)
    	at scala.Option.map(Option.scala:242)
    	at wvlet.airframe.surface.ClassMethodSurface.call(Surfaces.scala:429)
    	at wvlet.airframe.launcher.Launcher$.$anonfun$10$$anonfun$1(Launcher.scala:77)
    	at wvlet.airframe.launcher.CommandLauncher.execute$$anonfun$2(Launcher.scala:291)
    	at scala.Option.map(Option.scala:242)
    	at wvlet.airframe.launcher.CommandLauncher.execute(Launcher.scala:291)
    	at wvlet.airframe.launcher.Launcher.execute(Launcher.scala:143)
    	at wvlet.lang.cli.WvletREPLMain$.main$$anonfun$1(WvletREPLMain.scala:35)
    	at wvlet.lang.cli.WvletREPLMain$.main$$anonfun$adapted$1(WvletREPLMain.scala:35)
    	at scala.Function0.apply$mcV$sp(Function0.scala:42)
    	at wvlet.lang.cli.WvletREPLMain$.wrap(WvletREPLMain.scala:24)
    	at wvlet.lang.cli.WvletREPLMain$.main(WvletREPLMain.scala:35)
    	at wvlet.lang.cli.WvletREPLMain.main(WvletREPLMain.scala)
    	Suppressed: java.io.IOException: Unable to load terminal provider jansi: null
    		at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:427)
    		at org.jline.terminal.TerminalBuilder.checkProvider(TerminalBuilder.java:1065)
    		at org.jline.terminal.TerminalBuilder.getProviders(TerminalBuilder.java:1034)
    		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:708)
    		... 31 more
    	Caused by: java.lang.reflect.InvocationTargetException
    		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandl
    eAccessor.java:74)
    		at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
    		at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
    		at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:425)
    		... 34 more
    	Caused by: java.lang.NoClassDefFoundError: org/fusesource/jansi/internal/Kernel32
    		at org.jline.terminal.impl.jansi.JansiTerminalProvider.verifyAtLeast(JansiTerminalProvider.java:85)
    		at org.jline.terminal.impl.jansi.JansiTerminalProvider.<init>(JansiTerminalProvider.java:93)
    		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandl
    eAccessor.java:62)
    		... 37 more
    	Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.internal.Kernel32
    		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
    		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
    		... 40 more
    	Suppressed: java.io.IOException: Unable to load terminal provider jna: null
    		at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:427)
    		at org.jline.terminal.TerminalBuilder.checkProvider(TerminalBuilder.java:1065)
    		at org.jline.terminal.TerminalBuilder.getProviders(TerminalBuilder.java:1036)
    		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:708)
    		... 31 more
    	Caused by: java.lang.reflect.InvocationTargetException
    		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandl
    eAccessor.java:74)
    		at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
    		at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
    		at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:425)
    		... 34 more
    	Caused by: java.lang.NoClassDefFoundError: com/sun/jna/Platform
    		at org.jline.terminal.impl.jna.JnaNativePty.isatty(JnaNativePty.java:192)
    		at org.jline.terminal.impl.jna.JnaNativePty.isPosixSystemStream(JnaNativePty.java:170)
    		at org.jline.terminal.impl.jna.JnaTerminalProvider.checkSystemStream(JnaTerminalProvider.java:284)
    		at org.jline.terminal.impl.jna.JnaTerminalProvider.<init>(JnaTerminalProvider.java:31)
    		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandl
    eAccessor.java:62)
    		... 37 more
    	Caused by: java.lang.ClassNotFoundException: com.sun.jna.Platform
    		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
    		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
    		... 42 more  - (WvletREPLMain.scala:33)

```

--- a/wvlet-cli/src/main/scala/wvlet/lang/cli/WvletREPL.scala
+++ b/wvlet-cli/src/main/scala/wvlet/lang/cli/WvletREPL.scala
@@ -50,8 +50,9 @@ class WvletREPL(workEnv: WorkEnv, runner: WvletScriptRunner) extends AutoCloseab
   private val terminal = TerminalBuilder
     .builder()
     .name("wvlet-shell")
-    // Use dumb terminal for sbt testing
-    .dumb(WvletMain.isInSbt)
+    // Use dumb terminal for sbt testing or non-TTY environments (e.g., Claude Code, CI/CD)
+    // When TTY env var is not set, we're not in a real terminal
+    .dumb(WvletMain.isInSbt || sys.env.get("TTY").isEmpty)
     .build()
 
   private val historyFile = new File(workEnv.cacheFolder, ".wv_history")

--- a/wvlet-cli/src/main/scala/wvlet/lang/cli/WvletREPL.scala
+++ b/wvlet-cli/src/main/scala/wvlet/lang/cli/WvletREPL.scala
@@ -51,7 +51,8 @@ class WvletREPL(workEnv: WorkEnv, runner: WvletScriptRunner) extends AutoCloseab
     .builder()
     .name("wvlet-shell")
     // Use dumb terminal for sbt testing or non-TTY environments (e.g., Claude Code, CI/CD)
-    // When TTY env var is not set, we're not in a real terminal
+    // Note: We check TTY env var instead of System.console() == null because System.console()
+    // returns ProxyingConsole (not null) in Java 24+ even in non-TTY environments
     .dumb(WvletMain.isInSbt || sys.env.get("TTY").isEmpty)
     .build()
 

